dotcom-rendering/src/components/FollowButtons.tsx

```diff
@@ -65,74 +65,63 @@ const containerStyles = css`
 	column-gap: 0.2em;
 `;
 
-const notificationsTextSpan = ({ isFollowing }: Pick<Props, 'isFollowing'>) => (
+const notificationsTextSpan = ({
+	isFollowing,
+}: Pick<ButtonProps, 'isFollowing'>) => (
 	<span>Notifications {isFollowing ? 'on' : 'off'}</span>
 );
 
 const tagTextSpan = ({
 	isFollowing,
 	displayName,
-}: Pick<Props, 'isFollowing' | 'displayName'>) => (
+}: {
+	isFollowing: boolean;
+	displayName?: string;
+}) => (
 	<span>
 		{isFollowing ? 'Following' : 'Follow'} {displayName}
 	</span>
 );
 
-const generateFollowButton = ({
-	isFollowing,
-	onClickHandler,
-	iconIsFollowing,
-	iconIsNotFollowing,
-	textSpan,
-}: Props & {
-	iconIsFollowing: EmotionJSX.Element;
-	iconIsNotFollowing: EmotionJSX.Element;
-	textSpan: EmotionJSX.Element;
-}): EmotionJSX.Element => {
-	return (
-		<>
-			<button onClick={onClickHandler} type="button" css={buttonStyles}>
-				<span css={containerStyles}>
-					<FollowIcon
-						isFollowing={isFollowing}
-						iconIsFollowing={iconIsFollowing}
-						iconIsNotFollowing={iconIsNotFollowing}
-					/>
-					{textSpan}
-				</span>
-			</button>
-		</>
-	);
-};
-
 // bridget props
-type Props = {
-	displayName?: string;
+type ButtonProps = {
 	isFollowing: boolean;
 	onClickHandler: () => void;
 };
 
 export const FollowNotificationsButton = ({
 	isFollowing,
 	onClickHandler,
-}: Props) =>
-	generateFollowButton({
-		isFollowing,
-		onClickHandler,
-		textSpan: notificationsTextSpan({ isFollowing }),
-		iconIsFollowing: <SvgNotificationsOn size="xsmall" />,
-		iconIsNotFollowing: <SvgNotificationsOff size="xsmall" />,
-	});
+}: ButtonProps) => {
+	return (
+		<button onClick={onClickHandler} type="button" css={buttonStyles}>
+			<span css={containerStyles}>
+				<FollowIcon
+					isFollowing={isFollowing}
+					iconIsFollowing={<SvgNotificationsOn size="xsmall" />}
+					iconIsNotFollowing={<SvgNotificationsOff size="xsmall" />}
+				/>
+				{notificationsTextSpan({ isFollowing })}
+			</span>
+		</button>
```

Comment:
The UI for the source button doesn't match the designs. But perhaps with the new rounded icons and the "subdued" button option this could work. Again somethine we can revist with the wider byline changes.